### DISCUSSION TITLE
GeoMap: clamp view range to the Tile Server capabilities

### DISCUSSION THREAD

**Discussion by anurag-patnaik** [grafana/grafana#41505] [Dashboards, Panels & Visualizations]
**What happened**:
If the "Map View" -> "Initial View" panel has a zoom level set to more than the max zoom level configured for the Tile Server, the GeoMap panel loads up as blank.

This is because an HTTP 404 is returned from the tile server if the initial zoom level > max zoom level.

**What you expected to happen**:
1. Could we change the error response to an HTTP 400 Bad Request and return an appropriate reason rather than an HTTP 404. 
2. Gracefully degrade by showing an appropriate error in the GeoMap panel instead of a blank screen?

**How to reproduce it (as minimally and precisely as possible)**:
1. Set a max zoom level for GeoMap tile server
2. Create a GeoMap panel and set the "Map View" > "Initial View" > Zoom value to more than the max zoom value.
3. Load the dashboard. The panel will not render any tiles and remain blank.

<img width="379" alt="Initial View Zoom" src="https://user-images.githubusercontent.com/949305/140431333-76df5c6b-d2f4-43c2-9afd-bc0df97992d1.png">

If you try and ping the Tile Server directly with a zoom value > max, you can reproduce the HTTP 404:

_Sample Request_
https://some-tile-server-url.com/tiles/11/576/746.png

_HTTP 404 Error Response from Tile Server_
<img width="1023" alt="Chrome Dev Tools - 404 Error" src="https://user-images.githubusercontent.com/949305/140431380-fdbd7c37-af32-402d-a7b4-0249ff0b5702.png">

**Anything else we need to know?**:
Zooming out to a zoom level lower than max will make the tiles render again.

**Environment**:
- Grafana version: 8.2
- Data source type & version: Google Sheets
- OS Grafana is installed on: Linux
- User OS & Browser: macOS Big Sur 11.6 / Chrome
- Grafana plugins: GeoMap

**Comment by ryantxu** [comment]
Hi @anurag-patnaik -- I moved this to a discussion topic while we figure out how to actually solve the issue.  The core problem is that as is, we do not know the valid range of the tile server and so can not accuratly clip it.

In the case you show -- it seems like the value was intentionally saved at a resolution beyond what your tile server can produce, so I'm not really sure what the expected behavior would be.

**Comment by anurag-patnaik** [comment]
Hi @ryantxu. Thank you for the update. I appreciate that. 

And yes, the initial zoom value being set beyond what the tile server can produce is certainly user error. Maybe we can consider some UX changes to notify the user on why the dashboard panel doesn't render - in this case, due to an invalid initial zoom value configuration.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
